internal/worker/caasprober/controller_test.go

```diff
@@ -302,17 +308,56 @@ func (s *ControllerSuite) TestControllerProbeFailDetailed(c *gc.C) {
 			recorder := httptest.NewRecorder()
 			h.ServeHTTP(recorder, req)
 			c.Check(recorder.Result().StatusCode, gc.Equals, http.StatusInternalServerError)
-			c.Check(recorder.Body.String(), gc.Matches, `(?m)Internal Server Error: probe (liveness|readiness|startup)`)
+			c.Check(recorder.Body.String(), gc.Matches,
+				`(?m)Internal Server Error: probe (liveness|readiness|startup)\n\- test: test error`)
 			waitGroup.Done()
 			return nil
 		},
 		RemoveHandlerFunc: func(m, p string) {},
 	}
 
-	probeFail := probe.ProberFn(func() (bool, error) {
+	probeErr := probe.ProberFn(func() (bool, error) {
 		return false, errors.New("test error")
 	})
 
+	probes := caasprober.NewCAASProbes()
+	livenessAgg, _ := probes.ProbeAggregate(probe.ProbeLiveness)
+	livenessAgg.AddProber("test", probeErr)
+	readinessAgg, _ := probes.ProbeAggregate(probe.ProbeReadiness)
+	readinessAgg.AddProber("test", probeErr)
+	startupAgg, _ := probes.ProbeAggregate(probe.ProbeStartup)
+	startupAgg.AddProber("test", probeErr)
+	controller, err := caasprober.NewController(probes, &mux)
+	c.Assert(err, jc.ErrorIsNil)
+
+	waitGroup.Wait()
```

Comment:
As an aside, this could block forever...

We should use the general pattern of:

```sh
done := make(chan struct{})
go func() {
     waitGroup.Wait()
     close(done)
}()
select {
    case <-done:
    case <-time.After(testing.LongWait):
        c.Fatalf(".......")
}
```